.github/workflows/publish-release.yml

```diff
@@ -12,7 +12,7 @@ jobs:
   publish:
     name: Publish final artifacts
     if: github.repository == 'line/armeria'
-    runs-on: self-hosted
+    runs-on: self-hosted-for-non-pr
```

Comment:
optional; `self-hosted-safe`, `self-hosted-unsafe` might be more straightforward